src/main/java/com/example/demo/StudentService.java

```diff
@@ -16,27 +16,17 @@ public List<Student> getAllStudents() {
     }
 
     public Student getStudentById(String id) {
-        if (studentRepository.existsById(id)) {
-            Optional<Student> student = studentRepository.findById(id);
+            Optional<Student> student = studentRepository.getById(id);
             return student.orElse(null);
-        }
-        return null;
     }
 
     public Student updateStudent(String id, Student studentDetails) {
-        if (studentRepository.existsById(id)) {
             return studentRepository.save(studentDetails);
-        }
-        return null;
     }
 
-    public Student deleteStudent(String id) {
-        if (studentRepository.existsById(id)) {
-            Optional<Student> delStudent = studentRepository.findById(id);
-            studentRepository.deleteById(id);
-            return delStudent.orElse(null);
-        }
-        return null;
+    public boolean deleteStudent(String id) {
+            Optional<Student> delStudent = studentRepository.getById(id);
+            return studentRepository.deleteById(delStudent.get().getId());
```

Comment:
Is there a difference between User and Student objects that would indicate a difference in how they are deleted? I see that you handled the delete functions differently and wondered if there was a reason.

Reply:
Potentially. It depends on how dependent we want the student tables on the user tables. For example, if all students are users, you should not be able to delete a user that is a student. Or, another option is to ALSO delete the student record if the user related is a student.

Reply:
For the purpose of learning, would it be helpful to utilize multiple options for how we handled this? We could spread the functionality across different entity types where different types utilize different methods. Alternatively, we could just work on one option and once it worked, we move on to a different idea and implement that. What are your thoughts on what is best practice?

Reply:
My mantra is simpler is almost always better. 

Let's start simple, get it to work, write tests and update code to make it more robust, then move on to the next one.